src/utils.ts

```diff
@@ -1,7 +1,36 @@
 import type { Drop, DropSuffix } from './internals'
 
 type Digit = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9'
-type Separator = ' ' | '_' | '-' | '.' | '/'
+
+const SEPARATORS = [
+  ' ',
+  '_',
+  '-',
+  '.',
+  '/',
+  '|',
+  '(',
+  ')',
+  '[',
+  ']',
+  '{',
+  '}',
+] as const
+
+type Separator = (typeof SEPARATORS)[number]
+
+/** Escape characters with special significance in regular expressions */
+function escapeCharacter(char: string): string {
+  if (['[', ']', '{', '}', '(', ')', '|', '/', '-'].includes(char)) {
+    return `\\${char}`
+  }
+  return char
```

Comment:
I miss the '\' character.
Also, maybe you could have:
```ts
const UNESCAPED_SEPARATORS = ['[', ']'] as const // ...
const SEPARATORS = join([UNESCAPED_SEPARATORS, [' ', '-'] as const]) // ...

/** Escape characters with special significance in regular expressions */
function escapeCharacter(char: string): string {
  if (UNESCAPED_SEPARATORS.includes(char)) {
    return `\\${char}`
  }
  return char
}
```

Just to make sure the next time we touch this array we are adding a separator in the right category.